lib/openstax/aws/stack.rb

```diff
@@ -1,5 +1,58 @@
+require 'json'
+
 module OpenStax::Aws
   class Stack
+    class Event
+      def initialize(aws_stack_event)
+        @aws_stack_event = aws_stack_event
+      end
+            
+      def status
+        @aws_stack_event.data.resource_status
+      end
+            
+      def reason
+        @aws_stack_event.data.resource_status_reason
+      end
+            
+      def failure?
+        status.include?("_FAILED") || status.include?("ROLLBACK_COMPLETE")
+      end
+
+      def failure_with_reason?
+        status.include?("_FAILED") || status == "ROLLBACK_IN_PROGRESS"
+      end
+            
+      def user_initiated?
+        reason == "User Initiated"
+      end
+    end
+
+    class Status
+      def initialize(stack)
+        @stack = stack.aws_stack
+      end
+
+      def status
+        @stack.stack_status
+      end
+
+      def failure?
+        status.include?("_FAILED") || status.include?("ROLLBACK_COMPLETE")
+      end
+
+      def failure_with_reason?
+        status.include?("_FAILED") || status == "ROLLBACK_IN_PROGRESS"
+      end
+
+      def events
```

Comment:
These aren't `events`.  This is like the "to hash" method?  So `def to_h`?  And then like above, provide a `'def to_json; to_h.to_json; end` method.